### PR TITLE
s3: make client name configurable in s3 dependent implementations;

### PR DESCRIPTION
**Pull request by bogdanfinn**

### CODE REVIEW

File: pkg/blob/runner.go [hunk 1]
```diff
@@ -63,22 +64,22 @@ type batchRunner struct {
...
 	channels *BatchRunnerChannels
+	client   gosoS3.Client
+	metric   metric.Writer
 	settings *BatchRunnerSettings
 }
 
-func NewBatchRunner(ctx context.Context, config cfg.Config, logger log.Logger) (*batchRunner, error) {
+func NewBatchRunner(ctx context.Context, config cfg.Config, logger log.Logger) (BatchRunner, error) {
 	settings := &BatchRunnerSettings{}
 	config.UnmarshalKey("blob", settings)
```
Comment (applike-ss) [inline pkg/blob/runner.go:75]: i'd love to see something like 
```suggestion
func NewBatchRunner(ctx context.Context, config cfg.Config, logger log.Logger, name string) (BatchRunner, error) {
	settings := &BatchRunnerSettings{}
	config.UnmarshalKey(fmt.Sprint("blob.%s", name), settings)
```

File: pkg/blob/store.go [hunk 2]
```diff
@@ -83,11 +88,12 @@ type Store interface {
...
 
 type s3Store struct {
-	logger   log.Logger
+	logger log.Logger
+
+	bucket   *string
 	channels *BatchRunnerChannels
 	client   gosoS3.Client
 
-	bucket *string
```
Comment (applike-ss) [inline pkg/blob/store.go:90]: that's not in line with our coding policy, is it?

File: docs/fixtures/readme.md [hunk 3]
```diff
@@ -49,7 +49,7 @@ func main() {
 or for example when using the `BlobFixtureWriterFactory`:
 
 ```yaml
-blobstore:
+blob:
   blobconfig:
```
Comment (selm0) [inline docs/fixtures/readme.md:53]: Don't you need to add the client name here now? E.g.
```
blob:
  default:
    blobconfig:
```

Comment (bogdanfinn) [reply docs/fixtures/readme.md:53]: `blobconfig` is the clientname here :-D

File: pkg/fixtures/purger_blob.go [hunk 4]
```diff
@@ -22,7 +22,7 @@ func newBlobPurger(ctx context.Context, config cfg.Config, logger log.Logger, se
 		return nil, fmt.Errorf("can not create blob store: %w", err)
 	}
 
-	br, err := blob.NewBatchRunner(ctx, config, logger)
+	br, err := blob.NewBatchRunner(ctx, config, logger, "default")
```
Comment (applike-ss) [inline pkg/fixtures/purger_blob.go:25]:
```suggestion
	br, err := blob.NewBatchRunner(ctx, config, logger, settings.ConfigName)
```

File: pkg/fixtures/writer_blob.go [hunk 5]
```diff
@@ -41,7 +41,7 @@ func BlobFixtureWriterFactory(settings *BlobFixturesSettings) FixtureWriterFacto
 			return nil, fmt.Errorf("can not create blob store: %w", err)
 		}
 
-		br, err := blob.NewBatchRunner(ctx, config, logger)
+		br, err := blob.NewBatchRunner(ctx, config, logger, "default")
```
Comment (applike-ss) [inline pkg/fixtures/writer_blob.go:44]:
```suggestion
		br, err := blob.NewBatchRunner(ctx, config, logger, settings.ConfigName)
```

**Review by applike-ss — CHANGES_REQUESTED** [review]

**Review by applike-ss — APPROVED** [review]

**Pull request merged**